### PR TITLE
New setting: "display addon names in English"

### LINKED ISSUE #5186: New SA setting: "display addon names in English"
### Why this would be helpful/solve a problem

Until we can solve #2670 directly, we should allow users to see addon names, description, setting names, etc. in English, even if their browser language is set to a different language that we support.
Tags such as "Recommended" and other strings would remain in that language, but many users would still appreciate the feature.

### What this change does

A new setting available on "More Settings", which could be named "display addon names in English".
We could add a small "beta" tag next to it, to indicate that it's meant to be temporal. In the future, we should let the user set the whole SA UI to any of the supported languages, and it would affect everything except the extension description.

The value of this setting would be stored in `chrome.storage.local`, it would not be exportable through the JSON file, and it would set `useDefault` to `true` on this line: https://github.com/ScratchAddons/ScratchAddons/blob/master/background/load-addon-manifests.js#L42

Changing this setting would require reloading the extension, with exactly the same logic as importing a JSON settings file.

### Possible alternatives

Waiting until we can provide an actual language selector, and allow users to choose other languages, not just English.
But I've seen too many suggestions about this. I even saw users asking how to change the language on Reddit (r/scratch).

### Demonstration(s) and additional context

_No response_

### PR DESCRIPTION
**Pull request by WorldLanguages**
Resolves #5186

brave_oudFb3KiC2

### CODE REVIEW

File: _locales/en/messages.json [hunk 1]
```diff
@@ -308,6 +308,12 @@
   "confirmImport": {
     "message": "Confirm"
   },
+  "language": {
+    "message": "Language"
+  },
+  "applySettings": {
+    "message": "Apply settings"
```
Comment (lisa-wolfgang) [inline _locales/en/messages.json:315]:
```suggestion
    "message": "Apply changes"
```

**Review by apple502j — APPROVED** [review]

File: webpages/settings/index.html [hunk 2]
```diff
@@ -262,6 +262,34 @@ <h1 v-cloak>{{ msg("settings") }}</h1>
...
+          <div class="addon-settings">
+            <div class="addon-setting" style="margin-top: 0">
+              <input
+                type="checkbox"
+                class="setting-input check"
+                v-model="forceEnglishSetting"
+                style="margin-left: 0; margin-right: 8px"
+              />
+              <span>Show addon names and descriptions in English</span>
+              <div class="badge red">{{ "Beta" }}</div>
```
Comment (WorldLanguages) [inline webpages/settings/index.html:280]: TODO: localize "beta" (string already exists)

**Comment by WorldLanguages** [comment]
Note: the name for the setting is intentionally not localized, at the moment

**Comment by mxmou** [comment]
The switch should probably be a checkbox because it's in front of the label.

**Comment by WorldLanguages** [comment]
> The switch should probably be a checkbox because it's in front of the label.

Could you submit a PR? I'm pretty sure my design is not ideal, but I think it's worth it, because some users really wanted this feature.

**Comment by lisa-wolfgang** [comment]
Do we have checkboxes anywhere else in the extension anymore? I'm not sure why this specific scenario warrants an inconsistent design component.

**Comment by Samq64** [comment]
Or the switch could be moved to the right to be consistent with everything else.

**Comment by WorldLanguages** [comment]
Note: enabling this new setting has the side-effect of converting these strings to English too (popup names):

image

I wouldn't say this was intentional, but it's not a bug either.

**Pull request merged**